kai_analyzer_rpc/pkg/codec/stdio.go

```diff
@@ -0,0 +1,52 @@
+package codec
+
+import (
+	"io"
+
+	"github.com/cenkalti/rpc2"
+	"github.com/cenkalti/rpc2/jsonrpc"
+	"github.com/go-logr/logr"
+)
+
+// Connection wraps a pair of unidirectional streams as an io.ReadWriteCloser.
+type Connection struct {
+	Input  io.ReadCloser
+	Output io.WriteCloser
+	Logger logr.Logger
+}
+
+// Read implements io.ReadWriteCloser.
+func (c Connection) Read(p []byte) (n int, err error) {
+	r, err := c.Input.Read(p)
+	c.Logger.Info("read from connection", "bytes read", r, "error", err, "bytes", string(p))
```

Comment:
I wonder if we should bump this level because we are already logging this in the codec